okhttp-urlconnection/src/main/java/okhttp3/JavaNetCookieJar.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import okhttp3.internal.Util.delimiterOffset
+import okhttp3.internal.Util.trimSubstring
+import okhttp3.internal.cookieToString
+import okhttp3.internal.platform.Platform
+import okhttp3.internal.platform.Platform.Companion.WARN
+import java.io.IOException
+import java.net.CookieHandler
+import java.net.HttpCookie
+import java.util.ArrayList
+import java.util.Collections
+
+/** A cookie jar that delegates to a [java.net.CookieHandler].  */
+class JavaNetCookieJar(private val cookieHandler: CookieHandler) : CookieJar {
+
+  override fun saveFromResponse(url: HttpUrl, cookies: List<Cookie>) {
+    val cookieStrings = mutableListOf<String>()
+    for (cookie in cookies) {
+      cookieStrings.add(cookieToString(cookie, true))
+    }
+    val multimap = Collections.singletonMap<String, List<String>>("Set-Cookie", cookieStrings)
```

Comment:
`mapOf("Set-Cookie" to cookieStrings)`? I believe it has a single-argument version which calls this method

Reply:
Nice. Fixed.